python/ray/ml/preprocessors/scaler.py

```diff
@@ -92,4 +94,129 @@ def column_min_max_scaler(s: pd.Series):
+    range [-1, 1].
+
+    Args:
+        columns: The columns that will individually be scaled.
+    """
+
+    def __init__(self, columns: List[str]):
+        super().__init__()
+        self.columns = columns
+
+    def _fit(self, dataset: Dataset) -> Preprocessor:
+        aggregates = [AbsMax(col) for col in self.columns]
+        self.stats_ = dataset.aggregate(*aggregates)
+        return self
+
+    def _transform_pandas(self, df: pd.DataFrame):
+        def column_abs_max_scaler(s: pd.Series):
+            s_abs_max = self.stats_[f"abs_max({s.name})"]
+
+            # Handle division by zero.
+            # All values are 0.
+            if s_abs_max == 0:
+                s_abs_max = 1
+
+            return s / s_abs_max
+
+        df.loc[:, self.columns] = df.loc[:, self.columns].transform(
+            column_abs_max_scaler
+        )
+        return df
+
+    def __repr__(self):
+        stats = getattr(self, "stats_", None)
+        return f"<Scaler columns={self.columns} stats={stats}>"
+
+
+class RobustScaler(Preprocessor):
+    """Scale values within columns based on their quantile range.
+
+    For each column, each value will be transformed to
+    ``(value - median) / (high_quantile - low_quantile)``,
+    where ``median`` , ``high_quantile``, and ``low_quantile``
+    are calculated from the fitted dataset.
+
+    Args:
+        columns: The columns that will be scaled individually.
+        quantile_range: A tuple that defines the lower and upper quantile to scale to.
+                        Defaults to the 1st and 3rd quartiles: (0.25, 0.75).
+    """
+
+    def __init__(
+        self, columns: List[str], quantile_range: Tuple[float, float] = (0.25, 0.75)
+    ):
+        super().__init__()
+        self.columns = columns
+        self.quantile_range = quantile_range
+
+    def _fit(self, dataset: Dataset) -> Preprocessor:
+        low = self.quantile_range[0]
+        med = 0.50
+        high = self.quantile_range[1]
+
+        num_records = dataset.count()
+        max_index = num_records - 1
+        split_indices = [int(percentile * max_index) for percentile in (low, med, high)]
+
+        self.stats_ = {}
+
+        # TODO(matt): Handle case where quantile lands between 2 numbers.
+        # The current implementation will simply choose the closest index.
+        # This will affect the results of small datasets more than large datasets.
+        for col in self.columns:
+            filtered_dataset = dataset.map_batches(lambda df: df[[col]])
+            sorted_dataset = filtered_dataset.sort(col)
+            split_datasets = sorted_dataset.split_at_indices(split_indices)
+
+            def _get_first_value(ds: Dataset, c: str):
+                return ds.take(1)[0][c]
+
+            low_val = _get_first_value(split_datasets[1], col)
+            med_val = _get_first_value(split_datasets[2], col)
+            high_val = _get_first_value(split_datasets[3], col)
```

Comment:
Supernit: Unpacking might make this a bit cleaner/more clear.
```suggestion
            _, low, med, high = sorted_dataset.split_at_indices(split_indices)

            def _get_first_value(ds: Dataset, c: str):
                return ds.take(1)[0][c]

            low_val = _get_first_value(low, col)
            med_val = _get_first_value(med, col)
            high_val = _get_first_value(high, col)
```